src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -348,8 +358,23 @@ public function sendHeaders(): static
             header('Set-Cookie: '.$cookie, false, $this->statusCode);
         }
 
+        if ($statusCode) {
+            if (\function_exists('headers_send')) {
+                headers_send($statusCode);
+
+                return $this;
+            }
+
+            if ($statusCode >= 100 && $statusCode < 200) {
+                // skip informational responses if not supported by the SAPI
```

Comment:
shouldn't it also skip the `header()` calls in such case, to avoid adding duplicate headers on the final response ?

Reply:
Good catch. I fixed this logic. Headers must be added only one time for all responses.